app/renderer/components/preferences/syncTab.js

```diff
@@ -321,91 +396,440 @@ class SyncTab extends ImmutableComponent {
+            <img className={css(styles.syncOverlayBody__syncQRImg)}
+              src={this.props.syncData.get('seedQr')}
+              data-l10n-id='syncQRImg'
+              data-test-id='syncQRImg'
+            />
+          </Column>
+        </Grid>
+      </div>
+    )
+  }
+
+  get scanCodeOverlayFooter () {
+    return (
+      <div className={css(styles.syncOverlayFooter_split)}>
+        <BrowserButton secondaryColor
+          l10nId='syncTypeSecurityCode'
+          onClick={this.scanCodeOverlayNoCameraAvailable.bind(this)}
+        />
+        <div>
+          <BrowserButton groupedItem secondaryColor
+            l10nId='backWithArrow'
+            testId='cancelButton'
+            onClick={this.scanCodeOverlayPreviousAction.bind(this)}
+          />
+          <BrowserButton groupedItem primaryColor
+            l10nId={
+              this.props.syncData.get('devices').size < 2
+                ? 'syncLookingForDevice'
+                : 'done'
+              }
+            testId='syncLookingForDevice'
+            disabled={this.props.syncData.get('devices').size < 2}
+            onClick={this.scanCodeOverlayNextAction.bind(this)}
+          />
+        </div>
+      </div>
+    )
+  }
+
+  onHideAnySetupOverlay () {
+    // hide every setup modal
+    this.props.hideOverlay('syncScanCode')
+    this.props.hideOverlay('syncChainCode')
+    this.props.hideOverlay('syncQRPassphrase')
+    this.props.hideOverlay('syncAdd')
+
+    if (!this.isSetup) {
+      // cancel sync without warning as user didn't complete setup
+      this.onReset(false)
+    }
+  }
+
+  onHideAnyRemovalOverlay () {
+    this.props.hideOverlay('syncReset')
+    this.props.hideOverlay('syncRemove')
   }
 
-  get resetOverlayFooter () {
+  scanCodeOverlayNoCameraAvailable () {
+    // hide current modal
+    this.props.hideOverlay('syncScanCode')
+    // open chain code modal
+    this.props.showOverlay('syncChainCode')
+  }
+
+  scanCodeOverlayPreviousAction () {
+    // hide current modal
+    this.props.hideOverlay('syncScanCode')
+    // open previous modal
+    this.props.showOverlay('syncStart')
+  }
+
+  checkDeviceUpdatesFor (modalName) {
+    const devices = this.props.syncData.get('devices')
+    const isSetupCompleted = this.props.syncData.get('setupCompleted')
+
+    // if setup is completed there's no need to check for updates
+    if (isSetupCompleted) {
+      return
+    }
+
+    if (devices.isEmpty()) {
+      // Update modal after 5s to check if new device is already fetch
+      setTimeout(() => this.forceUpdate(), 5000)
+    }
+
+    // the only way to finish sync setup is
+    // to have >= 2 devices in the list
+    if (devices.size >= 2) {
+      appActions.syncSetupCompleted(true)
+      // close current modal
+      this.props.hideOverlay(modalName)
+    }
+  }
+
+  scanCodeOverlayNextAction () {
+    // close current modal
+    this.props.hideOverlay('syncScanCode')
+  }
+
+  /**
+   * QR and Passphrase Overlay
```

Comment:
Currently there is no text that indicates this QR code or passphrase should be kept secret. I am worried this will lead to users sharing their QR code or passphrase in support requests. They should not do this since the information is an encryption key. 

@bradleyrichter can you mock up a warning somewhere telling users to never share their code words or QR code?

Reply:
for reference this is what ledger says: 

![screen shot 2018-04-09 at 3 34 34 pm](https://user-images.githubusercontent.com/549654/38526539-9805591e-3c0b-11e8-819b-798abebeb491.png)

Reply:
mockup:
![image](https://user-images.githubusercontent.com/13509546/38535594-10d8b50c-3c39-11e8-8bca-62089db5975f.png)

Reply:
done in 327d4a6da2bbdbff3ffae2a3b0f7df3138be486f

<img width="733" alt="screen shot 2018-04-10 at 6 38 00 pm" src="https://user-images.githubusercontent.com/4672033/38585320-cf8edd76-3cef-11e8-9073-a214bf7503f3.png">
<img width="752" alt="screen shot 2018-04-10 at 6 29 24 pm" src="https://user-images.githubusercontent.com/4672033/38585321-cfb77826-3cef-11e8-94a8-12d10c81bce5.png">
<img width="728" alt="screen shot 2018-04-10 at 6 29 17 pm" src="https://user-images.githubusercontent.com/4672033/38585322-cfe0a1ec-3cef-11e8-94a7-e782db0c7684.png">

Reply:
second screenshot has the text in different position given the extra footer button eating space